demos/portal-embed-demo/public/js/ttt/audio-controller.js

```diff
@@ -0,0 +1,369 @@
+    get css() {
+        return `
+            :host {
+                width: 100%;
+                height: 80px;
+                background-color: slategray;
+                position: fixed;
+                bottom: 0px;
+                z-index: 999;
+                display: block;
+            }
+            .controllers {
+                margin: 0 auto;
+                height:80%;
+                width:180px;
+            }
+            .controller {
+                width: 60px;
+                height: 80%;
+                text-align: center;
+                float: left;
+            }
+            .controller img{
+                height: 50%;
+                cursor: pointer;
+            }
+            .flip-horizontal {
+                transform: scale(-1, 1);
+            }
+            .title {
+                width: 100%;
+                height: 20%;
+                text-align: center;
+                margin-top: 8px;
+                font-size: 0.8em;
+                color: #FFF;
+            }
+            .timeline {
+                width: 90%;
+                max-width: 550px;
+                border-top: 1px solid #FFF;
+                margin: 8px auto 3px auto;
+                cursor: pointer;
+            }
+            .axis{
+                width: 10px;
+                height: 10px;
+                background-color: #FFF;
+                position: relative;
+                left: 0px;
+                top: -5px;
+                border-radius: 50%;
+                cursor: pointer;
+            }
+            .prev {
+                opacity: 0.3;
+            }
+            .next {
+                opacity: 0.3;
+            }
+        `;
+    }
+
+    /**
+     * HTML template
+     */
+    get template() {
+        return `
+            <div class='title'></div>
+            <div class='timeline'>
+                <div class='axis'></div>
+            </div>
+            <div class='controllers'>
+                <div class='controller'>
+                    <img class='prev flip-horizontal' src='/img/next.png'>
+                </div>
+                <div class='controller'>
+                    <img class='play' src='/img/play.png'>
+                </div>
+                <div class='controller'>
+                    <img class='next' src='/img/next.png'>
+                </div>
+            </div>
+        `
+    }
+
+    /**
+     * Controlling play button
+     */
+    play() {
+        this.playButton.src = '/img/pause.png';
+        const startAxis = _ => {
+            // Start the progress bar
+            const left = this.timeline.offsetWidth;
+            this.axis.style.transition = `${this.durationSec}s linear`;
+            this.axis.style.left = `${left}px`;
+            this.isPlaying = true;
+        }
+        // https://developers.google.com/web/updates/2017/06/play-request-was-interrupted
+        const promise = this.audio.play();
+        if(promise !== undefined) {
```

Comment:
Since it seems we want to treat the no-promise and promise-resolved case equivalently, we can equivalently just write:

```javascript
Promise.resolve(this.audio.play()).then(_ => {
    // Start the progress bar.
});
```

Reply:
...!! This is much more readable!!